sites/all/modules/custom/training7/training7.rules.inc

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Implements hook_rules_event_info().
+ */
+function training7_rules_event_info() {
+  $items = array(
+    'training7_user_visited_lottery' => array(
+      'label' => t('User has visited lottery page'),
+      'group' => t('Training7'),
+      'access callback' => TRUE,
+    ),
+  );
+  return $items;
+}
+
+/**
+ * Implements hook_rules_condition_info().
+ */
+function training7_rules_condition_info() {
+  $items = array(
+    'training7_lottery_result' => array(
+      'label' => t('Lottery is won'),
+      'group' => t('Training7'),
+    ),
+  );
+  return $items;
+}
+
+/**
+ * Implements hook_rules_action_info().
+ */
+function training7_rules_action_info() {
+  $items = array(
+    'training7_show_lottery_win_message' => array(
+      'label' => t('Display a lottery win message'),
+      'group' => t('Training7'),
+      'access callback' => TRUE,
+    ),
+  );
+  return $items;
+}
+
+/**
+ * Callback function for rule condition.
+ *
+ * Generates random number and sends it by xmlprc to get lottery result(bool).
+ * Stores in cache data for Lottery Table block.
+ */
+function training7_lottery_result() {
+
+  $rand = rand(1, 8);
+  $server = url($GLOBALS['base_url'] . '/xmlrpc.php', array('external' => TRUE));
```

Comment:
Use https://api.drupal.org/api/drupal/developer%21globals.php/global/base_url/7.x instead of $GLOBALS['base_url']